integrations/aws/main.py

```diff
@@ -69,19 +73,37 @@ async def _handle_global_resource_resync(
 async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     if kind in iter(ResourceKindsWithSpecialHandling):
         return
+
     await update_available_access_credentials()
     is_global = is_global_resource(kind)
-    async for credentials in get_accounts():
-        if is_global:
-            async for batch in _handle_global_resource_resync(kind, credentials):
-                yield batch
-        else:
-            async for session in credentials.create_session_for_each_region():
-                try:
-                    async for batch in resync_cloudcontrol(kind, session):
-                        yield batch
-                except Exception:
-                    continue
+    semaphore = asyncio.BoundedSemaphore(MAX_CONCURRENT_TASKS)
```

Comment:
Is there anywhere else you can initiate this semaphore?

Reply:
the object isn't meant to be used by another function unless re-instantiated for some of use case, in my opinion that's the best place for it ?

Reply:
we can also instantiate somewhere else, but the point is, a single instance or object wouldn't be reusable for all other functions.